### PR TITLE
validate boolean fields correctly

### PR DESCRIPTION
**Pull request by samsamann**
This PR provides a new form type called 'strictBoolean'. This gives us the ability to handle boolean fields correcty. Now, a required boolean field can only have false or true as value.

### CODE REVIEW

File: src/Graviton/DocumentBundle/Form/DataTransformer/BooleanTransformer.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    public function reverseTransform($value)
+    {
+        if (!isset($this->propertyPath)) {
+            throw new RuntimeException('propertyPath must be set');
+        }
+
+        $originData = $this->submittedData;
+        if (is_array($this->submittedData) && preg_match('@^\[+[0-9]\]$@', $this->propertyPath)) {
+            $index = (int) trim($this->propertyPath, '[]');
+            $originData = $this->submittedData[$index];
```
Comment (lapistano) [inline src/Graviton/DocumentBundle/Form/DataTransformer/BooleanTransformer.php]: even, if »submittedData« is an array it is not said there is an element named as the value of $index. So $originData will be null and the content valid.

File: src/Graviton/DocumentBundle/Form/Type/StrictBooleanType.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+     *
+     * @param FormBuilderInterface $builder The form builder
+     * @param array                $options The options
+     * @return void
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $dataTransformer = clone $this->transformer;
+        $dataTransformer->setPropertyPath($builder->getPropertyPath());
+        $dataTransformer->setSubmittedData($options['submitted_data']);
```
Comment (lapistano) [inline src/Graviton/DocumentBundle/Form/Type/StrictBooleanType.php]: $options['submitted_data'] might not exist and the script will proceed only logging a Notice. Is this ok?

Comment (samsamann) [inline src/Graviton/DocumentBundle/Form/Type/StrictBooleanType.php]: even, if $options['submitted_data'] is not set on the type, this option will still exist with ``null``. But I agree the default value is not very good.

File: src/Graviton/DocumentBundle/Form/DataTransformer/BooleanTransformer.php [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        $this->submittedData = $submittedData;
+    }
+
+    /**
+     * @inheritDoc
+     *
+     * @param mixed $value The value in the original representation
+     * @return mixed The transformed value
+     */
+    public function transform($value)
```
Comment (lapistano) [inline src/Graviton/DocumentBundle/Form/DataTransformer/BooleanTransformer.php]: from the parents DocBlock: 
> By convention, transform() should return an empty string if NULL is passed.

I think the transformer should honour that.

File: src/Graviton/DocumentBundle/Form/DataTransformer/BooleanTransformer.php [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        return $value;
+    }
+
+    /**
+     * @inheritDoc
+     *
+     * @param mixed $value The value in the transformed representation
+     * @return mixed The original value
+     */
+    public function reverseTransform($value)
```
Comment (lapistano) [inline src/Graviton/DocumentBundle/Form/DataTransformer/BooleanTransformer.php]: from the parents DocBlock: 
> By convention, transform() should return an empty string if NULL is passed.

I think the transformer should honour that.

**Pull request merged**